archunit/src/main/java/com/tngtech/archunit/library/freeze/TextFileBasedViolationStore.java

```diff
@@ -140,15 +148,38 @@ public boolean contains(ArchRule rule) {
     @Override
     public void save(ArchRule rule, List<String> violations) {
         log.trace("Storing evaluated rule '{}' with {} violations: {}", rule.getDescription(), violations.size(), violations);
+        if (violations.isEmpty() && warnEmptyRuleViolation) {
+            throw new StoreEmptyException(String.format("Saving empty violations for freezing rule is disabled (enable by configuration %s.%s=true)",
+                    ViolationStoreFactory.FREEZE_STORE_PROPERTY_NAME, WARN_EMPTY_RULE_VIOLATION_PROPERTY_NAME));
+        }
+        if (violations.isEmpty() && deleteEmptyRule && !contains(rule)) {
```

Comment:
another way to write this check would be 
```suggestion
        if (violations.isEmpty() && (deleteEmptyRule || skipEmptyRule) && !contains(rule)) {
```
where `skipEmptyRule` condition is only used for skipping creation of empty rule file. To keep the configuration simple, I decided to use `deleteEmptyRule` for both skipping file creation and deleting the empty file.